layers/cmd/valueset.go

```diff
@@ -321,27 +332,33 @@ func (vsets Valuesets) Lookup(ctx *ls.Context, req ls.ValuesetLookupRequest) (ls
 	if counter == 1 {
 		return results[resultIdx], nil
 	}
+	vsets.cache.Set(req, found)
 	return found, nil
 }
 
 type valuesetMarshal struct {
 	Valueset
-	Services      map[string]string      `json:"services" yaml:"services"`
-	Spreadsheets  []string               `json:"spreadsheets" yaml:"spreadsheets"`
-	Sets          []Valueset             `json:"valuesets" yaml:"valuesets"`
-	DatabaseFiles []string               `json:"databaseFiles" yaml:"databaseFiles"`
-	Databases     map[string]interface{} `json:"databases" yaml:"databases"`
+	Services      map[string]string        `json:"services" yaml:"services"`
+	Spreadsheets  []string                 `json:"spreadsheets" yaml:"spreadsheets"`
+	Sets          []Valueset               `json:"valuesets" yaml:"valuesets"`
+	DatabaseFiles []string                 `json:"databaseFiles" yaml:"databaseFiles"`
+	Databases     []map[string]interface{} `json:"databases" yaml:"databases"`
 }
 
 func LoadValuesetFiles(ctx *ls.Context, vs *Valuesets, files []string) error {
 	if vs.Sets == nil {
 		vs.Sets = make(map[string]Valueset)
 		vs.Services = make(map[string]string)
+		cache, err := NewValuesetCache()
```

Comment:
Pass the cache to LoadValuesFiles
If the passed cache is nil, have a no-cache implementation, and set that.